hw/mcu/stm/stm32_common/src/hal_nvreg.c

```diff
@@ -30,23 +32,38 @@
 #define HAL_NVREG_MAX (0)
 #endif
 
+#if MYNEWT_VAL_MCU_STM32F1
+#define HAL_NVREG_WIDTH_BYTES 2
+#define HAL_NVREG_START_INDEX 1
+#else
 /* RTC backup registers are 32-bits wide */
 #define HAL_NVREG_WIDTH_BYTES (4)
+#define HAL_NVREG_START_INDEX 0
+#endif
 
 void
 hal_nvreg_write(unsigned int reg, uint32_t val)
 {
 #if PWR_ENABLED
     RTC_HandleTypeDef hrtc = { .Instance = RTC };
     if (reg < HAL_NVREG_MAX) {
+#if defined(__HAL_RCC_PWR_IS_CLK_ENABLED)
+        bool disable_pwr = __HAL_RCC_PWR_IS_CLK_ENABLED();
```

Comment:
this looks like there is `!` missing in condition,  ie to disable only if it wasn't already enabled

also, for parity I'd put both enable() and disable() calls under same condition